geomesa/src/main/scala/geotrellis/spark/io/geomesa/GeoMesaLayerReader.scala

```diff
@@ -0,0 +1,51 @@
+package geotrellis.spark.io.geomesa
+
+import geotrellis.geotools._
+import geotrellis.spark._
+import geotrellis.vector._
+
+import org.apache.accumulo.core.client.mapreduce.InputFormatBase
+import org.apache.hadoop.io.Text
+import org.apache.hadoop.mapreduce.Job
+import org.apache.spark.rdd.RDD
+import org.apache.spark.SparkContext
+import org.geotools.data._
+import org.locationtech.geomesa.jobs.mapreduce.GeoMesaInputFormat
+import org.locationtech.geomesa.jobs.GeoMesaConfigurator
+import org.opengis.feature.simple.{SimpleFeature, SimpleFeatureType}
+
+import scala.reflect.ClassTag
+
+class GeoMesaLayerReader(val attributeStore: GeoMesaAttributeStore, table: String)(implicit sc: SparkContext) extends Serializable {
```

Comment:
This should just take `DataStore` as constructor argument and provide `apply` overloads to help with construction. Also layer writer is generally for rasters. I would call this `GeoWaveFeatureReader`, @lossyrob thoughts?